feat: Add init api tokens option (#1181)
Unleash/unleash: website/docs/deploy/configuring-unleash.md

    - `demo` - Only requires an email to sign-in (was default in v3)
  - customAuthHandler: (function) - custom express middleware handling authentication. Used when type is set to `custom`
  - createAdminUser: (boolean) - whether to create an admin user with default password - Defaults to `true`

>   - initApiTokens: (ApiTokens[]) - Array of Api Tokens to create on startup. Will only be applied if there does not exist any API tokens in the database from before. 

Reviewer: ```suggestion
  - initApiTokens: (ApiTokens[]) - Array of API Tokens to create on startup. Will only be applied if there are no existing API tokens in the database already. 
```

>   - initApiTokens: (ApiTokens[]) - Array of Api Tokens to create on startup. Will only be applied if there does not exist any API tokens in the database from before. 

Reviewer 2: I will add these improvements.

>   - initApiTokens: (ApiTokens[]) - Array of Api Tokens to create on startup. Will only be applied if there does not exist any API tokens in the database from before. 

Reviewer 2: fixed in ce1cc8f3a8bbcab568c1b93685dcb1f9640cb656

      Example:
     ```ts
     [{
        environment: '*',
        project: '*',
        secret: '*:*:964a287e1b728cb5f4f3e0120df92cb5',
        type: ApiTokenType.ADMIN,
        username: 'some-user',

>     }]

Reviewer 2: The secret is the thing actually shared with the API client, thus duplicated. By prefixing it with project and environment it gives context, but we currently only support that for "Client Tokens", while admin tokens must be wild-carded.

>     }]

Reviewer 2: fyi: there is not actual requirements that the secret itself is prefixed, but we do that for new tokens to add context to the user of the token.

>     }]

Reviewer 2: (because the holder of an API token secret would be able to use the token and that way discover its capabilites, and you should take good care of it, I see no harm in adding the context in to the secret itself).

>     }]

Reviewer 3: I think I see what you mean. I was just thinking that it seems redundant and that you can easily supply unmatching values unless there is a check. But that's on the software design side, and you certainly have more insight into that than I do, so I trust you've taken these things into consideration. Thanks for the explanation! 🙏🏼

>     }]

Reviewer 2: When creating these from the API/UI, which is the normal user-flow, you will not be able to set the secret yourself. This is very specific way of short-circuit it all, and thus in find it OK to not build to much additional logic for this scenario.

     ```

>      You may also use the environment variable `INIT_ADMIN_API_TOKENS` which takes a comma separated list of Api Tokens to initialize. Example value of this variable could be: "`*:*:some-random-string, *:*:some-other-token`". PS! It is required that all Admin tokens target all environments and projects.

Reviewer: ```suggestion
     You may also use the environment variable `INIT_ADMIN_API_TOKENS` which takes a comma separated list of API Tokens to initialize. An example value for this variable could be: "`*:*:some-random-string, *:*:some-other-token`". NB: All Admin tokens must target all environments and projects.
```

>      You may also use the environment variable `INIT_ADMIN_API_TOKENS` which takes a comma separated list of Api Tokens to initialize. Example value of this variable could be: "`*:*:some-random-string, *:*:some-other-token`". PS! It is required that all Admin tokens target all environments and projects.

Reviewer 3: What's the actual format of the new API tokens? Is it `project:environment:hash` or `project:environment.hash`? The tokens I create (on Unleash 4.4.4) seem to take the latter form. Are both allowed?

Is the `INIT_ADMIN_API_TOKENS` env var equal to using the `initApiTokens` property? It carries a lot less data. If the actual token string is all that's necessary, why isn't the `initApiTokens` just a `string[]`?

Also, can you use spaces in comma-separated lists for env vars? I always assumed you had to have _only_ a comma between the entries, but I don't think I've ever checked. For instance: `a, b` vs `a,b`.

```suggestion
    You can also use the environment variable `INIT_ADMIN_API_TOKENS`. This variable should be set to a comma-separated list of API tokens to initialize (for instance `*:*:some-random-string, *:*:some-other-token`). All admin tokens **must** target all environments and projects.
```

Reviewer 2: 1. The format is `project:environment:hash`
2. _INIT_ADMIN_API_TOKENS env var equal to using the initApiTokens property?_ No not really, when using unleash via code you can define any token type to be initialized, while if you are just running Unleash and using the _INIT_ADMIN_API_TOKENS_ env variable, you will only be able to specify a list of Admin API tokens (and we will set the type to admin and define a username for you).
3. A single space is optional so both "a,b" and "a, b" is accepted, we have defined this where we parse the string via a simple regex like this `/,\s?/`

Reviewer 2: Language improvements added in ce1cc8f3a8bbcab568c1b93685dcb1f9640cb656

Reviewer 3: 1. That's what I thought. Does that mean you need to change the API keys you get from the instance? I just copied this one from the demo instance and it uses `project:env.string`: `demo-app:production.614a75cf68bef8703aa1bd8304938a81ec871f86ea40c975468eabd6`
2. Ah, so you can you create non-admin API tokens with the `apiTokens` property, then? I may have misunderstood that.
3. Got it! TIL 🤓

Reviewer 2: 2. Jup, any token type is accepted, and we even export the types for it so most developers using a modern IDE will get type  assistance.

Reviewer 2: 1. I assume that is a "client" token, which can be scoped to an environment and project.

Reviewer 3: 1. Correct! I don't have access to create tokens on the demo instance, but here's an admin token from my local OSS instance with the same pattern: `*:*.dcfab32f502d257cb153ceb2d27e3dcef61d6218bb182522a110362b`
2. Got it! I think that could use some clarification. I'll create a PR for that 🙋🏼
